.github/workflows/ci_pullrequest.yml

```diff
@@ -292,6 +292,9 @@ jobs:
       - name: "Setup base app_imports for mock"
         working-directory: ./nautobot/ui
         run: "echo 'export const NautobotApps = {}; export default NautobotApps;' > src/app_imports.js"
+      - name: "Setup base app_routes.json for test"
+        working-directory: ./nautobot/ui
+        run: "echo '{}' > src/app_routes.json"
```

Comment:
Are these steps that a developer needs to do before running `invoke unittest-ui` locally? If so let's make sure we have that well documented.

Are these steps that a user needs to do when installing Nautobot from pypi? Same concern.

Reply:
@glennmatthews What if instead we added said file in the repo bit `.gitignore` and ignored all changes?

So:
```
echo "{}" > nautobot/ui/src/app_routes.json
git add -f nautobot/ui/src/app_routes.json
```

Reply:
That's probably fine, I think?

Reply:
Womp womp...

> Editing .gitignore will only ignore files so they will not be added to the git repository. However files that are tracked already will not be ignored.

Testing locally it still picked up future changes to the file. I'll update the PR to create the file if it doesn't exist.

Reply:
https://github.com/nautobot/nautobot/pull/3736/files#diff-5c50d475237a0b724fe8d90c7862d19b2a48d8cf8f0c334b8e7d8d336db219f8R6-R9

How about this instead @glennmatthews?

Reply:
I don't have a strong opinion about how we do this, just want to make sure that if it's something a developer needs to do, they are informed about it. :-)

Reply:
They shouldn't have to do that anymore. Exception handling FTW!